File: src/api/schemas/mediakit.py
"""Pydantic schemas for advertiser-readable mediakit endpoint (B.5.1)."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MediakitAdvertiserResponse(BaseModel):
    """Advertiser-readable mediakit view.

    Excludes internal control fields (is_published flag, owner_user_id, counters).
    Returned only when underlying ChannelMediakit.is_published is True;
    otherwise the route returns 404 (no draft-existence leak).
    """

    description: str | None = None
    audience_description: str | None = None
    logo_file_id: str | None = None
    theme_color: str | None = None
    avg_post_reach: int
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: mini_app/src/api/mediakit.ts
import { api } from './client'

export interface MediakitAdvertiserResponse {
  description: string | null
  audience_description: string | null
  logo_file_id: string | null
  theme_color: string | null
  avg_post_reach: number
  updated_at: string
}

/**
 * Fetch advertiser-readable mediakit for a channel.
 *
 * Backend: GET /api/channels/{channelId}/mediakit (B.5.1).
 * Returns 200 with body if mediakit is_published=true; 404 otherwise
 * (parity for not-published / not-exists / no-mediakit — no draft leak).
 * Caller hook translates 404 into "not available" empty state.
 */
export function getChannelMediakit(channelId: number): Promise<MediakitAdvertiserResponse> {
  return api.get(`channels/${channelId}/mediakit`).json<MediakitAdvertiserResponse>()
}
